src/Modules/ArticlesList.jsx

```diff
@@ -7,47 +7,86 @@ export default function ArticlesList() {
 
     const [articlesList, setArticlesList] = useState([])
-    let urlString = "https://nc-news-78g8.onrender.com/api/articles"
     const [isLoading, setIsLoading] = useState(false)
     const [isError, setIsError] = useState("")
 
     const handleTopic = (e) => {
         setTopicQuery(e.target.value)
         setSearchParams({ topic: e.target.value });
     }
+    const handleSortBy = (e) => {
+        setSortByQuery(e.target.value)
+        setSearchParams({ sort_by: e.target.value });
+    }
+    const handleOrder = (e) => {
+        setOrderQuery(e.target.value)
+        setSearchParams({ order: e.target.value });
+    }
+
 
     useEffect(()=>{
+        let urlString = "https://nc-news-78g8.onrender.com/api/articles"
+        let prevQuery = false;
+
         setIsLoading(true)
 
+        if(topicQuery.length === 0 && sortByQuery.length === 0 && orderQuery.length === 0) {
+            urlString = "https://nc-news-78g8.onrender.com/api/articles"
+            navigate(`/articles`)
+        }
+
         if (topicQuery.length !== 0) {
-            urlString += `?topic=${topicQuery}`
-            navigate(`/articles?topic=${encodeURIComponent(topicQuery)}`);
+            if (!prevQuery) { urlString += `?`; prevQuery=true } else { urlString += `&` }
+            urlString += `topic=${topicQuery}`
         }
-        else {
-            urlString = "https://nc-news-78g8.onrender.com/api/articles"
-            navigate(`/articles`);
+        if (sortByQuery.length !== 0) {
+            if (!prevQuery) { urlString += `?`; prevQuery=true } else { urlString += `&` }
+            urlString += `sort_by=${sortByQuery}`
+        }
+        if (orderQuery.length !== 0) {
+            if (!prevQuery) { urlString += `?`; prevQuery=true } else { urlString += `&` }
+            urlString += `order=${orderQuery}`
         }
 
+        console.log(urlString)
+
+
+
         fetch(urlString)
         .then(response => response.json())
         .then(data => setArticlesList(data.articles))
         .then(()=> setIsLoading(false))
-    }, [topicQuery])
+    }, [topicQuery, sortByQuery, orderQuery])
 
     if (articlesList.length === 0) return <h1>Loading...</h1>
 
     else return (
-        <>
-            <label htmlFor="topics"><h2>Topic:</h2></label>
-            <select name="topic" id="topic" onChange={handleTopic}>
-                <option value="">Select a topic...</option>
-                <option value="cooking">Cooking</option>
-                <option value="coding">Coding</option>
-                <option value="football">Football</option>
-            </select>
+        <>  
+            <div className="query-bar">
+                <label htmlFor="topics"><h3>Topic:</h3></label>
+                <select name="topic" id="topic" className="query-dropdown" onChange={handleTopic}>
+                    <option value="">All</option>
+                    <option value="cooking">Cooking</option>
+                    <option value="coding">Coding</option>
```

Comment:
if we add a new topic this would cause an issue here - we could get all topics then map through to get these values. Depends if you back end allows adding a new topic